src/client/proxy/handlers/common.ts

```diff
@@ -282,9 +282,7 @@ const tryFulfillZhtmlRequest = async (
         renderer = new Renderer({rootDirId: cfg.remoteRootDirId} as any);
     }
 
-    const contentTypeFromExt = getContentTypeFromExt(ext || '');
-    const contentType = contentTypeFromExt || 'text/html';
-    res.header('Content-Type', contentType);
+    res.header('Content-Type', 'text/html');
```

Comment:
This assumption doesn't seem correct. Even though the function is called `tryFulfillZhtmlRequest`, in `templateFilename` argument it actually receives a route that matches the requested path. As long as we currently don't put any restrictions on values added in the `routes.json` file, a route may resolve to a file of any type, and the `tryFulfillStaticRequest` call will miss it. Therefore we still need to check the content type here, and probably prevent rendering if it is something other than `text/html`.

Reply:
Wait, how can that happen? We are returning `renderer.render`, which has `Twig` under the hood. Also, we call `.toString()` then. Is there any chance it can render an image or a binary?